modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/system/CPDefinitionSystemObjectDefinitionManager.java

```diff
@@ -63,6 +66,30 @@ public long addBaseModel(User user, Map<String, Object> values)
 		return product.getProductId();
 	}
 
+	@Override
+	public void checkPermission(
+			PermissionChecker permissionChecker, long primaryKey,
+			String actionId, long baseModelPrimaryKey)
+		throws PortalException {
+
+		CPDefinition cpDefinition = _cpDefinitionLocalService.fetchCPDefinition(
+			primaryKey);
+
+		if (cpDefinition == null) {
+			cpDefinition =
+				_cpDefinitionLocalService.fetchCPDefinitionByCProductId(
+					primaryKey);
+		}
+
+		if (cpDefinition == null) {
+			throw new NoSuchCPDefinitionException(
+				"No CPDefinition found for id = " + primaryKey);
+		}
+
+		_commerceCatalogPermission.check(
+			permissionChecker, cpDefinition.getCommerceCatalog(), actionId);
```

Comment:
I prefer to do the same as the method `_checkCommerceCatalogByCPDefinitionId`, then refactor:
```
CommerceCatalog commerceCatalog =
	_commerceCatalogLocalService.fetchCommerceCatalogByGroupId(
		cpDefinition.getGroupId());

_commerceCatalogModelResourcePermission.check(
	getPermissionChecker(), commerceCatalog, actionId);
```